### PR TITLE
chore(deps): update typescript to v5.6

### PR DESCRIPTION
**Pull request by SoonIter**
## Summary
chore(deps): update ts 5.6
close https://github.com/web-infra-dev/rspack/pull/7211

after https://github.com/web-infra-dev/rspack/pull/8072, we don't rely on tsc to generate js files, so we can upgrade the version of typescript.

1. upgrade and unify the ts version `^5.6.3`
2. add renovate to revert https://github.com/web-infra-dev/rspack/pull/5470/files

## Checklist

- [x] Tests updated (or not required).
- [x] Documentation updated (or not required).

### CODE REVIEW

File: api-extractor.json [hunk 1]
```diff
@@ -143,16 +143,22 @@
...
+     * DEFAULT VALUE: "[ "complete" ]"
+     */
+    "reportVariants": ["complete"],
+
     /**
      * The filename for the API report files.  It will be combined with "reportFolder" or "reportTempFolder" to produce
      * a full file path.
      *
      * The file extension should be ".api.md", and the string should not contain a path separator such as "\" or "/".
+     * API Extractor will automatically append an appropriate file extension such as .api.md
```
Comment (SoonIter) [inline api-extractor.json:156]: this a breaking change of `api-extractor`
before: `api.md`
now: `<package-name>(.beta?).api.md`

File: packages/rspack-test-tools/etc/test-tools.api.md [hunk 2]
```diff
@@ -4,10 +4,6 @@
 
 ```ts
 
-/// <reference types="../jest.d.ts" />
```
Comment (SoonIter) [inline packages/rspack-test-tools/etc/test-tools.api.md:7]: I do not know if this is the change of `api-extractor`, it does not matter I think

**Comment by SoonIter** [comment]
A noteworthy point is that `rspack-cli` is still using tsc and will preserve `await import` after this change.

We recommend using node18 and above, so I don't think there will be any problems.

@chenjiahan cc

**Review by chenjiahan — APPROVED** [review]

**Comment by chenjiahan** [comment]
> A noteworthy point is that rspack-cli is still using tsc and will preserve await import after this change.

LGTM, and `@rspack/cli` will switch to Rslib soon

**Pull request merged**